tests/opentelemetry-test-utils/src/opentelemetry/test/test_base.py

```diff
@@ -85,6 +91,63 @@ def sorted_spans(self, spans):  # pylint: disable=R0201
             reverse=True,
         )
 
+    def sorted_metrics(self, metrics):  # pylint: disable=R0201
+        """
+        Sorts metrics by metric name.
+        """
+        return sorted(
+            metrics,
+            key=lambda m: m.name,
+        )
+
+    def get_sorted_metrics(self):
+        resource_metrics = (
+            self.memory_metrics_reader.get_metrics_data().resource_metrics
+        )
+
+        all_metrics = []
+        for metrics in resource_metrics:
+            for scope_metrics in metrics.scope_metrics:
+                all_metrics.extend(scope_metrics.metrics)
+
+        return self.sorted_metrics(all_metrics)
+
+    def assert_metric_expected(
+        self,
+        metric: Metric,
+        expected_value: Union[int, float],
```

Comment:
The way I would find this useful is if it takes the current metric and the expected metric, not a single float/int value. The following code has things that don't make it useable for many use cases; why does the count have to be 1? How can one use this when the count is not 1? There are no assertions of min, max etc.?